internal/memory/store.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2022 Lekko Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package memory
+
+import (
+	"fmt"
+	"sync"
+
+	backendv1beta1 "buf.build/gen/go/lekkodev/cli/protocolbuffers/go/lekko/backend/v1beta1"
+	featurev1beta1 "buf.build/gen/go/lekkodev/cli/protocolbuffers/go/lekko/feature/v1beta1"
+	"github.com/lekkodev/go-sdk/pkg/eval"
+	"github.com/pkg/errors"
+	"google.golang.org/protobuf/proto"
+)
+
+var (
+	ErrConfigNotFound error = fmt.Errorf("config not found")
+)
+
+func newStore() *store {
+	return &store{
+		configs: make(map[configKey]configData),
+	}
+}
+
+type configKey struct {
+	namespace, config string
+}
+
+type configData struct {
+	config    *featurev1beta1.Feature
+	configSHA string
+}
+
+// store implements an in-memory store for configurations. It stores
+// the commit sha of the contents. It supports atomically updating
+// all the configs with the contents of a new commit via the Update method.
+type store struct {
+	sync.RWMutex
+	configs   map[configKey]configData
```

Comment:
i recall complex typed keys for maps being far slower than something like a string for both `get` and `set` operations. maybe it's been fixed in newer versions of Go but maybe it's worth a benchmark.

Reply:
i added a benchmark in `store_test.go`. turns out `map[string]map[string]..` is ~40% faster than `map[configKey]...`, so I updated the implementation to that